exchanges/huobi/huobi_websocket.go

```diff
@@ -119,129 +121,210 @@ func (h *HUOBI) wsMultiConnectionFunnel(ws *wshandler.WebsocketConnection, url s
+	err := json.Unmarshal(respRaw, &init)
 	if err != nil {
-		h.Websocket.DataHandler <- err
-		return
+		return err
+	}
+	if init.Subscribed != "" ||
+		init.UnSubscribed != "" ||
+		init.Op == "sub" ||
+		init.Op == "unsub" {
+		// TODO handle subs
+		return nil
 	}
 	if init.Ping != 0 {
 		h.sendPingResponse(init.Ping)
-		return
+		return nil
 	}
 	if init.ErrorMessage == "api-signature-not-valid" {
 		h.Websocket.SetCanUseAuthenticatedEndpoints(false)
-	}
-	if init.Op == "sub" {
-		if h.Verbose {
-			log.Debugf(log.ExchangeSys, "%v: %v: Successfully subscribed to %v", h.Name, resp.URL, init.Topic)
-		}
-		return
+		return errors.New(h.Name + " - invalid credentials. Authenticated requests disabled")
 	}
 	if init.ClientID > 0 {
-		h.AuthenticatedWebsocketConn.AddResponseWithID(init.ClientID, resp.Raw)
-		return
+		if h.WebsocketConn.IsIDWaitingForResponse(init.ClientID) {
+			h.WebsocketConn.SetResponseIDAndData(init.ClientID, respRaw)
+			return nil
+		}
 	}
 
 	switch {
 	case strings.EqualFold(init.Op, authOp):
 		h.Websocket.SetCanUseAuthenticatedEndpoints(true)
-		var response WsAuthenticatedDataResponse
-		err := json.Unmarshal(resp.Raw, &response)
+		err := json.Unmarshal(respRaw, &init)
 		if err != nil {
-			h.Websocket.DataHandler <- err
+			return err
 		}
-		h.Websocket.DataHandler <- response
+		h.Websocket.DataHandler <- init
+
 	case strings.EqualFold(init.Topic, "accounts"):
 		var response WsAuthenticatedAccountsResponse
-		err := json.Unmarshal(resp.Raw, &response)
+		err := json.Unmarshal(respRaw, &response)
 		if err != nil {
-			h.Websocket.DataHandler <- err
+			return err
 		}
 		h.Websocket.DataHandler <- response
+
 	case strings.Contains(init.Topic, "orders") &&
 		strings.Contains(init.Topic, "update"):
 		var response WsAuthenticatedOrdersUpdateResponse
-		err := json.Unmarshal(resp.Raw, &response)
+		err := json.Unmarshal(respRaw, &response)
 		if err != nil {
-			h.Websocket.DataHandler <- err
+			return err
 		}
-		h.Websocket.DataHandler <- response
+		data := strings.Split(response.Topic, ".")
+		if len(data) < 2 {
+			return errors.New(h.Name + " - currency could not be extracted from response")
+		}
+		orderID := strconv.FormatInt(response.Data.OrderID, 10)
+		var oSide order.Side
+		oSide, err = stringToOrderSide(response.Data.OrderType)
+		if err != nil {
+			h.Websocket.DataHandler <- order.ClassificationError{
+				Exchange: h.Name,
+				OrderID:  orderID,
+				Err:      err,
+			}
+		}
+		var oType order.Type
+		oType, err = stringToOrderType(response.Data.OrderType)
+		if err != nil {
+			h.Websocket.DataHandler <- order.ClassificationError{
+				Exchange: h.Name,
+				OrderID:  orderID,
+				Err:      err,
+			}
+		}
+		var oStatus order.Status
+		oStatus, err = stringToOrderStatus(response.Data.OrderState)
+		if err != nil {
+			h.Websocket.DataHandler <- order.ClassificationError{
+				Exchange: h.Name,
+				OrderID:  orderID,
+				Err:      err,
+			}
+		}
+		p := currency.NewPairFromString(data[1])
```

Comment:
Please use currency.NewPairFromFormattedPairs here otherwise this will cause pair mismatches